### PR TITLE
Move diagnostic about `@_weakLinked` not working for COFF into the type checker

### PR DESCRIPTION
**Pull request by DougGregor**
This allows us to use `@_weakLinked` inside of a `#if` properly.

### CODE REVIEW

**Comment by slavapestov** [comment]
One less annoying thing for ASTGen to deal with!

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: lib/Sema/TypeCheckAttr.cpp [hunk 1]
```diff
@@ -7342,6 +7342,13 @@ void AttributeChecker::visitStaticExclusiveOnlyAttr(
   }
 }
 
+void AttributeChecker::visitWeakLinkedAttr(WeakLinkedAttr *attr) {
+  if (Ctx.LangOpts.Target.isOSBinFormatCOFF()) {
+    diagnoseAndRemoveAttr(attr, diag::attr_unsupported_on_target,
+                          attr->getAttrName(), Ctx.LangOpts.Target.str());
+  }
```
Comment (compnerd) [inline lib/Sema/TypeCheckAttr.cpp:7349]:
```suggestion
  if (!Ctx.LangOpts.Target.isOSBinFormatCOFF())
    return;
  diagnoseAndRemoveAttr(attr, diag::attr_unsupported_on_target,
                        attr->getAttrName(), Ctx.LangOpts.Target.str());
```

**Review by compnerd — APPROVED** [review]

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

**Pull request merged**